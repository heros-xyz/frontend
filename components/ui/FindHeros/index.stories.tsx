import type { Meta, StoryObj } from "@storybook/react";
import FindHeros from ".";

const meta: Meta<typeof FindHeros> = {
  title: "Fan/FindHeros",
  component: FindHeros,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof FindHeros>;

export const FindHeros1: Story = {};
