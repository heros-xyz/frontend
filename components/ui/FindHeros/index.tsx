import {
  Box,
  BoxProps,
  Center,
  Flex,
  HStack,
  Input,
  InputGroup,
  InputLeftElement,
  Text,
} from "@chakra-ui/react";
import React, { useCallback, useState } from "react";
import { useRouter } from "next/router";
import Link from "next/link";
import { LogoMiniIcon } from "@/components/svg/LogoMini";
import { FindIcon } from "@/components/svg/Find";
import { useSearchAthleteProfileQuery } from "@/api/athlete";
import useDebounce from "@/hooks/useDebounce";
import { IconInfo } from "@/components/svg/IconInfo";
import SearchSuggestionsList from "../SearchSuggestions/List";

interface IFindHeros extends BoxProps {
  value?: string;
  onChange?: (el: React.ChangeEvent<HTMLInputElement>) => void;
  showResult?: boolean;
  onSeeAll?: () => void;
}

const FindHeros: React.FC<IFindHeros> = ({ value, onSeeAll, ...props }) => {
  const [searchValue, setSearchValue] = useState("");
  const [isSearchBarFocused, setFocus] = useState(false);
  const [showSuggestList, setShowSuggestList] = useState(false);

  const router = useRouter();

  const onChange = useCallback((el: React.ChangeEvent<HTMLInputElement>) => {
    setSearchValue(el.target.value);
    setShowSuggestList(true);
  }, []);

  const searchValueDebounced = useDebounce(searchValue, 500);

  const LIMIT = 5;

  const { data } = useSearchAthleteProfileQuery(
    {
      searching: searchValueDebounced?.toLocaleLowerCase(),
      limit: LIMIT,
    },
    { skip: searchValue.length <= 1 }
  );

  const onShowAllResult = () => {
    setShowSuggestList(false);
    if (router.pathname !== "/fan/all-result") {
      router.push({ pathname: "/fan/all-result", query: { searchValue } });
      return;
    }

    onSeeAll && onSeeAll();
  };

  const onFocus = () => setFocus(true);
  const onBlur = () => setFocus(false);
  return (
    <Box {...props} position="relative">
      <Flex h={14} pt={1}>
        <Link href={"/"}>
          <Center
            h={{ base: 6, xl: "1.875rem" }}
            w={{ base: 6, xl: "1.875rem" }}
            my={3}
            mr={3}
          >
            <LogoMiniIcon w="100%" h="100%" />
          </Center>
        </Link>

        <InputGroup>
          <InputLeftElement pointerEvents="none">
            <FindIcon color="white" />
          </InputLeftElement>
          <Input
            type="text"
            placeholder="Find your heros"
            _placeholder={{ color: "grey.100" }}
            _focusVisible={{ boxShadow: "none", outline: "none" }}
            color="white"
            border={0}
            value={value}
            borderRadius={0}
            borderBottom="1px"
            onChange={onChange}
            onFocus={onFocus}
            onBlur={onBlur}
            fontSize={{ base: "md", xl: "lg" }}
            onKeyUp={(e) => {
              if (e.key === "Enter" && searchValue.length > 1) {
                onShowAllResult();
                setShowSuggestList(false);
              }
            }}
          />
        </InputGroup>
      </Flex>
      {data && showSuggestList && (
        <SearchSuggestionsList
          zIndex={15}
          right={0}
          left={8}
          top={{ base: "50px", lg: "60px" }}
          position="absolute"
          buttonName={data.length ? "See All Results" : "No Result Found"}
          items={data}
          onShowAllResult={onShowAllResult}
          onClick={() => setShowSuggestList(false)}
        />
      )}
      {searchValue.length === 0 && isSearchBarFocused && (
        <HStack
          bg="acccent.4"
          p={{ base: 3, lg: 4 }}
          borderRadius="base"
          left={8}
          right={0}
          position="absolute"
          zIndex={15}
          alignItems="start"
        >
          <IconInfo
            w={{ base: "15px", lg: "24px" }}
            h={{ base: "15px", lg: "24px" }}
            mt="0.5"
            mr={{ lg: 2 }}
          />
          <Text fontSize={{ base: "xs", lg: "md" }} fontWeight={500}>
            You can either search by athletesâ€™ name or their sports and sports
            related terms.
          </Text>
        </HStack>
      )}
    </Box>
  );
};

export default FindHeros;
