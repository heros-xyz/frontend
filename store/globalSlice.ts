import { PayloadAction, createSlice } from "@reduxjs/toolkit";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  isLoading: false,
  isMute: true,
  isPlaying: false,
  accessToken: '',
  refreshToken: ''
};

export const appState = createSlice({
  name: "appState",
  initialState: initialState,
  reducers: {
    startLoading: (state) => {
      state.isLoading = true;
    },
    finishLoading: (state) => {
      state.isLoading = false;
    },
    playVideo: (state) => {
      state.isPlaying = true;
    },
    pauseVideo: (state) => {
      state.isPlaying = false;
    },
    setAccessToken: (state, action: PayloadAction<string>) => {
      state.accessToken = action.payload
    },
    setRefreshToken: (state, action: PayloadAction<string>) => {
      state.refreshToken = action.payload
    }
  },
  extraReducers: {
    [HYDRATE]: (state, action) => {
      if (action.payload.appState.accessToken) {
        state.accessToken = action.payload.appState.accessToken;
      }

      if (action.payload.appState.refreshToken) {
        state.refreshToken = action.payload.appState.refreshToken;
      }
    }
  }
});

export const { startLoading, finishLoading, setAccessToken, setRefreshToken } = appState.actions;

export default appState.reducer;
