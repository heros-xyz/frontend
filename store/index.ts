import { configureStore } from "@reduxjs/toolkit";
import { createWrapper } from "next-redux-wrapper";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { setupListeners } from "@reduxjs/toolkit/query";
import globalSlice from "./globalSlice";

export const store = configureStore({
  reducer: {
    appState: globalSlice,

  },
  devTools: true,
  middleware: (gDM) => {
    return gDM().concat(

    );
  },
});

const makeStore = () => store;

export type AppStore = ReturnType<typeof makeStore>;
export type AppState = ReturnType<AppStore["getState"]>;
export type AppDispatch = AppStore["dispatch"];

export const wrapper = createWrapper<AppStore>(makeStore);
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<AppState> = useSelector;

setupListeners(store.dispatch);
