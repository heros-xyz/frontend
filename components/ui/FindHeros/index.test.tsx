import { render } from "@testing-library/react";
import mockRouter from "next-router-mock";
import { Provider } from "react-redux";
import { store } from "@/store";
import FindHeros from ".";

jest.mock("next/router", () => require("next-router-mock"));

it("renders homepage unchanged", () => {
  mockRouter.push("/initial-path");
  const { container } = render(
    <Provider store={store}>
      <FindHeros />
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
