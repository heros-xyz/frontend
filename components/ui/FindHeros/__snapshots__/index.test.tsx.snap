// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <div
      class="css-13co1gz"
    >
      <div
        class="css-eafid9"
      >
        <svg
          class="chakra-icon css-nug9sg"
          focusable="false"
          max-height="24"
          max-width="24"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.4241 4.57087C20.3021 3.14491 18.8708 1.99252 17.2382 1.20069C15.6056 0.408859 13.8145 -0.00170108 12 5.29732e-06C8.42126 5.29732e-06 5.72835 0.809597 3.52935 3.29421C1.65676 5.41053 0 8.95169 0 12C0 15.4091 1.48175 18.5218 3.7645 20.7058C5.91947 22.7674 10.194 24 13.412 24C17.3386 24 19.9284 21.5047 22.1178 18.5884C23.621 16.5838 24 14.7004 24 12C24.0037 9.30382 23.0956 6.68561 21.423 4.57087M15.3125 14.2409C13.3797 17.0938 10.3346 18.4027 8.50501 17.1646C6.67538 15.9266 6.75591 12.611 8.6854 9.75591C10.6149 6.90086 13.6632 5.59414 15.4928 6.83215C17.3225 8.07015 17.2419 11.3858 15.3125 14.2409Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="chakra-input__group css-bx0blc"
      >
        <div
          class="chakra-input__left-element css-1u2k0bl"
        >
          <svg
            class="chakra-icon css-1jhziqh"
            focusable="false"
            viewBox="0 0 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.5 16.5L11.5 11.5M13.1667 7.33333C13.1667 10.555 10.555 13.1667 7.33333 13.1667C4.11167 13.1667 1.5 10.555 1.5 7.33333C1.5 4.11167 4.11167 1.5 7.33333 1.5C10.555 1.5 13.1667 4.11167 13.1667 7.33333Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </div>
        <input
          class="chakra-input css-16cgzid"
          placeholder="Find your heros"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
